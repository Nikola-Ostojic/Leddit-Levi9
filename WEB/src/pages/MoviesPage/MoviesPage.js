import React, { useEffect, useState } from "react";
import { Container, LinearProgress } from "@material-ui/core";
import { withRouter } from "react-router-dom";

import MovieItem from "components/MovieItem";
import MoviesFilter from "components/MoviesFilter";
import CreateMovie from "components/CreateMovie";

export const MoviesPage = ({
  moviesData: { data, loading, page, totalPages, totalItems },
  error,
  fetchMovies,
  history,
  userRole
}) => {
  const [movieName, setMovieName] = useState("");
  const [itemsPerPage, setItemsPerPage] = useState(5);
  const [currentPage, setCurrentPage] = useState(page || 1);

  // This effect will get triggered if any of the properties
  // used for filtering are changed, furthermore, also triggered on the first render
  useEffect(() => {
    fetchMovies(movieName, currentPage, itemsPerPage);
  }, [movieName, currentPage, itemsPerPage, fetchMovies]);

  // When changing items per page, we always want to start from the first page
  useEffect(() => {
    setCurrentPage(1);
  }, [itemsPerPage]);

  if (error) {
    return <p>{error}</p>;
  }

  if (!data || loading) {
    return (
      <Container>
        <LinearProgress />
      </Container>
    );
  }

  return (
    <Container>
      <MoviesFilter
        totalPages={totalPages}
        movieName={movieName}
        setMovieName={setMovieName}
        currentPage={currentPage}
        setCurrentPage={setCurrentPage}
        itemsPerPage={itemsPerPage}
        setItemsPerPage={setItemsPerPage}
      />
      Items found: {totalItems}
      {data?.map((m) => (
        <MovieItem key={m.id} movie={m} onClick={history.push} />
      ))}
      {userRole === "Admin" && (
        <CreateMovie
          loadMovies={() => fetchMovies(movieName, currentPage, itemsPerPage)}
        />
      )}
    </Container>
  );
};
export default withRouter(MoviesPage);
