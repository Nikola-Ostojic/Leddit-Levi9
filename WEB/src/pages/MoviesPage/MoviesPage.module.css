.filterContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.formControl {
  width: 120px;
}
