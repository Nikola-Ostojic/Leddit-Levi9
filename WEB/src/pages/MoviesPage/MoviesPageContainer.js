import { connect } from "react-redux";

import { fetchMovies } from "modules/Movies/MoviesActions";
import {
  getMoviesData,
  getMoviesError,
  getMoviesLoading
} from "modules/Movies/MoviesSelectors";
import { getUserRole } from "modules/Auth/AuthSelectors";

import MoviesPage from "./MoviesPage";

export const mapStateToProps = (state) => ({
  moviesData: getMoviesData(state),
  error: getMoviesError(state),
  loading: getMoviesLoading(state),
  userRole: getUserRole(state)
});

export default connect(mapStateToProps, { fetchMovies })(MoviesPage);
