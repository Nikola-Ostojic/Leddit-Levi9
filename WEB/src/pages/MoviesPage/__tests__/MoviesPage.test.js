import React from "react";
import { shallow } from "enzyme";

import MoviesPage from "../MoviesPage";

describe("MoviesPage tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<MoviesPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
