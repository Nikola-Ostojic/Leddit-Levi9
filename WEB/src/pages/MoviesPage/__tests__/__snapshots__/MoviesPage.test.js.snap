// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPage tests Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
